src/core/Database.cpp

```diff
@@ -308,6 +308,22 @@ void Database::recycleGroup(Group* group)
      }
 }
 
+void Database::emptyRecycleBin()
+{
+    if (m_metadata->recycleBinEnabled() && m_metadata->recycleBin()) {
+        // destroying direct entries of the recycle bin
+        QList<Entry*> subEntries = m_metadata->recycleBin()->entries();
+        for (Entry* entry : subEntries) {
+            delete entry;
```

Comment:
Some additional information. After cleaning the recycle bin and saving, a KDBX file is reduced in size and can be opened in KeePass 2.35 (the original .NET program). The latter doesn't report any errors and the recycle bin looks clean.